### PR TITLE
Add details for determining the parent Span from a Context

### PR DESCRIPTION
**Pull request by mwear**
[OTEP 66](https://github.com/open-telemetry/oteps/blob/master/text/0066-separate-context-propagation.md#default-span-parentage) has a section where it offers guidance on how to parent a `Span` given a `Context`. This PR integrates those details into the api-tracing spec.

The bulk of this PR is the following text:

#### Parenting from a Context

When a new `Span` is created from a `Context` there are several scenarios to
consider when selecting a parent. A `Context` can contain:

- A current `Span`
- An extracted `SpanContext`
- A current `Span` and an extracted `SpanContext`
- Neither a current `Span` or an extracted `Span` context

In order handle these scenarios, a convention has been established for assigning
a parent from a `Context`. The precedence for parent selection is as follows:

- Use the current `Span`.
- Use the extracted `SpanContext`.
- There is no parent. Create a root `Span`.

### CODE REVIEW

File: specification/api-tracing.md [hunk 1]
```diff
@@ -260,6 +261,11 @@ The API MUST accept the following parameters:
 - The parent Span or parent Span context, and whether the new `Span` should be a
   root `Span`. API MAY also have an option for implicit parent context
   extraction from the current context as a default behavior.
+- The parent `Span` or a `Context` containing a parent `Span` or `SpanContext`,
```
Comment (yurishkuro) [inline specification/api-tracing.md]: did you mean to delete the previous bullet?

File: specification/api-tracing.md [hunk 2]
```diff
@@ -290,6 +296,23 @@ created in another process. Each propagators' deserialization must set
 `IsRemote` to true on a parent `SpanContext` so `Span` creation knows if the
 parent is remote.
 
+#### Parenting from a Context
+
+When a new `Span` is created from a `Context` there are several scenarios to
+consider when selecting a parent. A `Context` can contain:
```
Comment (yurishkuro) [inline specification/api-tracing.md]: Can shorten:

When a new `Span` is created from a `Context`, the `Context` may contain:

File: specification/api-tracing.md [hunk 3]
```diff
@@ -290,6 +296,23 @@ created in another process. Each propagators' deserialization must set
...
+
+- A current `Span`
+- An extracted `SpanContext`
+- A current `Span` and an extracted `SpanContext`
+- Neither a current `Span` or an extracted `Span` context
+
+In order handle these scenarios, a convention has been established for assigning
+a parent from a `Context`. The precedence for parent selection is as follows:
```
Comment (yurishkuro) [inline specification/api-tracing.md]: Can shorten:

The parent should be selected in the following order of precedence:

File: specification/api-tracing.md [hunk 4]
```diff
@@ -290,6 +296,23 @@ created in another process. Each propagators' deserialization must set
...
+
+- Use the current `Span`.
```
Comment (yurishkuro) [inline specification/api-tracing.md]:
```suggestion
- Use the current `Span`, if available.
```

File: specification/api-tracing.md [hunk 5]
```diff
@@ -290,6 +296,23 @@ created in another process. Each propagators' deserialization must set
...
+- Use the extracted `SpanContext`.
```
Comment (yurishkuro) [inline specification/api-tracing.md]:
```suggestion
- Use the extracted `SpanContext`, if available.
```

File: specification/api-tracing.md (hunk 1 above)
Comment (mwear) [reply specification/api-tracing.md]: 🤦‍♂

**Comment by mwear** [comment]
I should have batched the suggestions, but thanks for the review @yurishkuro.

File: specification/api-tracing.md [hunk 6]
```diff
@@ -20,6 +20,7 @@ Table of Contents
 * [SpanContext](#spancontext)
 * [Span](#span)
   * [Span creation](#span-creation)
+    * [Parenting from a Context](#parenting-from-a-context)
```
Comment (arminru) [inline specification/api-tracing.md]: I'm not a native speaker but _parenting from a context_ sounds odd to me and I wouldn't know right away what this is about. So far I've only heard of parenting as in raising children.
How about
- determining the parent span from a context
- setting the parent span from a context
- creating child spans from a context

or something like that?

File: specification/api-tracing.md [hunk 7]
```diff
@@ -290,6 +293,21 @@ created in another process. Each propagators' deserialization must set
 `IsRemote` to true on a parent `SpanContext` so `Span` creation knows if the
 parent is remote.
 
+#### Parenting from a Context
+
+When a new `Span` is created from a `Context` the `Context` may contain:
```
Comment (arminru) [inline specification/api-tracing.md]:
```suggestion
When a new `Span` is created from a `Context`, the `Context` may contain:
```

File: specification/api-tracing.md (hunk 6 above)
Comment (jmacd) [reply specification/api-tracing.md]: I'm happy with "determining the parent span from a context".

File: specification/api-tracing.md [hunk 8]
```diff
@@ -257,9 +258,11 @@ as a separate operation.
 The API MUST accept the following parameters:
 
 - The span name. This is a required parameter.
-- The parent Span or parent Span context, and whether the new `Span` should be a
-  root `Span`. API MAY also have an option for implicit parent context
-  extraction from the current context as a default behavior.
+- The parent `Span` or a `Context` containing a parent `Span` or `SpanContext`,
+  and whether the new `Span` should be a root `Span`. API MAY also have an
+  option for implicit parenting from the current context as a default behavior.
+  See [Parenting from a Context](#parenting-from-a-context) for guidance on
```
Comment (dyladan) [inline specification/api-tracing.md]: Broken link

Comment (mwear) [reply specification/api-tracing.md]: 🤦‍♂, thanks!

**Pull request merged**